packages/dart_frog_cli/lib/src/commands/dev/dev.dart

```diff
@@ -103,12 +109,14 @@ class DevCommand extends DartFrogCommand {
   Future<int> run() async {
     _ensureRuntimeCompatibility(cwd);
 
+    final host = (results['host'] as String?) ?? _defaultHost;
```

Comment:
You already defined `defaultsTo` to `_defaultHost`, therefore `results['host']` will never be null.